MailCore/Cache/DraftContentManager.swift

```diff
@@ -46,6 +47,34 @@ public class DraftContentManager: ObservableObject {
             shouldAddSignatureText: draftBodyResult.shouldAddSignatureText,
             attachments: draftBodyResult.attachments
         )
+
+        return try await signature
+    }
+
+    public func updateSignature(with newSignature: Signature) {
+        let realm = mailboxManager.getRealm()
+        guard let liveIncompleteDraft = realm.object(ofType: Draft.self, forPrimaryKey: incompleteDraft.localUUID) else {
+            return
+        }
+
+        do {
+            let parsedBody = try SwiftSoup.parse(liveIncompleteDraft.body)
+            // If we find the previous signature, we replace it with the new one
+            // otherwise we append the signature at the end of the document
+            if let foundSignatureDiv = try parsedBody.select(".\(Constants.signatureWrapperIdentifier)").first {
```

Comment:
What happens if I create a draft from the web, then try to change on mobile ?
(does the web wraps the signature in the same div as we do ?)

Reply:
Tested, it works

Reply:
Discussed IRL, it works right now as the Web uses the same identifier as we do. All good.